flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/util/UpdatingPlanChecker.scala

```diff
@@ -116,14 +135,100 @@ object UpdatingPlanChecker {
           val windowStartEnd = w.getWindowProperties.map(_.name)
           // we have only a unique key if at least one window property is selected
           if (windowStartEnd.nonEmpty) {
-            keys = Some(groupKeys ++ windowStartEnd)
+            val smallestAttribute = windowStartEnd.sorted.head
```

Comment:
Replace with `windowStartEnd.min`. Can you explain this logic? Selecting only the first string looks a bit weird.

Reply:
Thanks, `windowStartEnd.min` is better, i want to get the lexicographic smallest attribute from `windowStartEnd` .  I have sorted it before get the first string.